### PR TITLE
1274 Add minus operation for set difference to support CORE-000334 expected-variable checks

### PR DESCRIPTION
**Pull request by RakeshBobba03**
Issue #1274 asked for a way to compare output variables in validation reports. Rather than adding a special "compared" flag in reporting, we chose to add a `minus` operation so rules can compute set differences themselves and expose them as output variables. That gives the engine a single, consistent mechanism: rules produce comparison results via operations, and the existing output-variable reporting surfaces them.

This PR adds a `minus` operation that implements that set-difference logic. It takes `name` (minuend) and `value` (subtrahend) as references to other operations and returns elements in `name` that are not in `value`, keeping the order from `name`. When `value` is empty or missing, it returns all elements from `name`.

Implementation changes include a new `Minus` operation class with `_normalize_to_list` and `_set_difference_preserve_order` helpers; an optional `value` field on `OperationParams`; passing `value` from the rule processor into `OperationParams`; registering `minus` in the operations factory; and updating the Operations JSON schema to define and require the `value` property for the minus operator. Operations.md documents the operation with an example. Unit tests cover the helper logic, main behavior, both PandasDataset and DaskDataset, the empty-value case, and the missing-name-ref case.

Files used for testing:
[CORE-000334.yaml](https://github.com/user-attachments/files/25244049/CORE-000334.yaml)
[unit-test-coreid-CG0016-negative.xlsx](https://github.com/user-attachments/files/25244050/unit-test-coreid-CG0016-negative.xlsx)

### CODE REVIEW

File: cdisc_rules_engine/operations/operations_factory.py
Comment (RamilCDISC) [inline cdisc_rules_engine/operations/operations_factory.py:1]: Could you please set this file. it seems as there is no much changes but for some reason whole file is in diff. It is difficult to review like this.

Comment (RakeshBobba03) [reply cdisc_rules_engine/operations/operations_factory.py:1]: I made the update, it was a formatting issue

File: tests/unit/test_operations/test_minus.py
Comment (RamilCDISC) [inline tests/unit/test_operations/test_minus.py:1]: Could you please add test cases for duplicate value and scalars. Also for the case where value in the list is empty string ""

File: cdisc_rules_engine/operations/minus.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def _execute_operation(self):
+        name_ref = self.params.target  # minuend (list A)
+        value_ref = self.params.value  # subtrahend (list B)
+
+        if not name_ref or name_ref not in self.evaluation_dataset.columns:
+            return []
+        list_a = self.evaluation_dataset[name_ref].iloc[0]
+        if not value_ref or value_ref not in self.evaluation_dataset.columns:
+            # listA minus empty = all of listA (per Sam)
```
Comment (RamilCDISC) [inline cdisc_rules_engine/operations/minus.py]: Please remove the name form comment. If you would like to highlight for Sam, you can send him a message.

Comment (RakeshBobba03) [reply cdisc_rules_engine/operations/minus.py]: sorry, those were meant for me when making changes, must've missed to take them out. Taking them out ASAP

File: resources/schema/rule/Operations.md [hunk 2]
```diff
@@ -750,6 +750,34 @@ Operations:
     operator: get_column_order_from_dataset
 ```
 
+### minus
+
+Computes set difference: elements in `name` that are not in `value`. Uses [set difference](<https://en.wikipedia.org/wiki/Set_(mathematics)#Set_difference>) semantics (A ∖ B). Preserves order from the first list. Both `name` and `value` must reference other operation results (e.g., `$expected_variables`, `$dataset_variables`).
+
+When `value` is empty or missing, returns all elements from `name`.
```
Comment (SFJohnson24) [inline resources/schema/rule/Operations.md]: why is this line separate from the rest of the operation description?

File: resources/schema/rule/Operations.md [hunk 3]
```diff
@@ -750,6 +750,34 @@ Operations:
...
+Operations:
+  - id: $expected_variables
+    operator: expected_variables
+  - id: $dataset_variables
+    operator: get_column_order_from_dataset
+  - id: $expected_minus_dataset
+    name: $expected_variables
+    operator: minus
+    value: $dataset_variables
+Check:
```
Comment (SFJohnson24) [inline resources/schema/rule/Operations.md]: this example is a bit bloated.  Can we only show the operation yaml and add a line to the description that says simply. 'Can be computed and added to output variables to display missing elements in error results'

File: cdisc_rules_engine/models/operation_params.py [hunk 4]
```diff
@@ -57,6 +57,7 @@ class OperationParams:
     regex: str = None
     returntype: str = None
     target: str = None
+    value: str = None
```
Comment (SFJohnson24) [inline cdisc_rules_engine/models/operation_params.py]: you will also need to make a PR to editor for the reference guide for the parameter you are adding to keep our docs up to date.  https://github.com/cdisc-org/conformance-rules-editor/blob/main/docs/check_parameter.md   this has all the parameters and a description of them.

File: resources/schema/rule/Operations.md (hunk 2 above)
Comment (RakeshBobba03) [reply resources/schema/rule/Operations.md]: This was by accident. Merged into a single paragraph: the “When subtract is empty or missing, returns all elements from name” wording is now part of the main minus description, and the whole section uses subtract instead of value.

File: resources/schema/rule/Operations.md (hunk 3 above)
Comment (RakeshBobba03) [reply resources/schema/rule/Operations.md]: Updated: the minus example now shows only the Operations YAML block, and the description ends with: “Can be computed and added to output variables to display missing elements in error results.”

File: cdisc_rules_engine/models/operation_params.py (hunk 4 above)
Comment (RakeshBobba03) [reply cdisc_rules_engine/models/operation_params.py]: Done. docs/check_parameter.md in the conformance-rules-editor repo now documents the subtract operation parameter (with description and minus example). PR: https://github.com/cdisc-org/conformance-rules-editor/pull/293.

Comment (SFJohnson24) [reply cdisc_rules_engine/models/operation_params.py]: you still need to get rid of value since we are replacing it subtract

Comment (RakeshBobba03) [reply cdisc_rules_engine/models/operation_params.py]: Removed value from OperationParams, from the rule processor’s OperationParams construction, and from the Operations JSON schema.

**Comment by SFJohnson24** [comment]
@RakeshBobba03 can you resolve the merge conflict.
check_parameter was also moved to engine.  https://github.com/cdisc-org/conformance-rules-editor/pull/293 I closed this--can you just move the text from it into https://github.com/cdisc-org/cdisc-rules-engine/blob/main/resources/schema/rule/check_parameter.md this file
we can get this merged then
